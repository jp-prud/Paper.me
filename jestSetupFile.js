import 'react-native-gesture-handler/jestSetup';
import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';

jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);

jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');

  Reanimated.default.call = () => {};

  return Reanimated;
});

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

export const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => {
      return {
        navigate: mockedNavigate,
        goBack: jest.fn(),
      };
    },
  };
});
