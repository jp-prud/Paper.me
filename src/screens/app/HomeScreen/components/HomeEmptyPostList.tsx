import {useNavigation} from '@react-navigation/native';

import {Box, Button, Text} from '@components';

export function HomeEmptyPostList() {
  const {navigate} = useNavigation();

  function handleCreatePost() {
    navigate('PreviewPostScreen');
  }

  return (
    <Box testID="home-empty-post-list">
      <Text textAlign="center">Nenhum post encontrado.</Text>
      <Box>
        <Text textAlign="center">Seja o primeiro a criar um post.</Text>
        <Button text="Criar post" onPress={handleCreatePost} />
      </Box>
    </Box>
  );
}
