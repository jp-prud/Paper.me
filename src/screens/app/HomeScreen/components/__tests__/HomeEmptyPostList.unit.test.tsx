import {fireEvent, render} from '@tests';
import {mockedNavigate} from 'jestSetupFile';

import {HomeEmptyPostList} from '../HomeEmptyPostList';
describe('HomeEmptyPostList - unit', () => {
  it('should render the component', () => {
    const {getByTestId, toJSON} = render(<HomeEmptyPostList />);

    expect(getByTestId('home-empty-post-list')).toBeDefined();
    expect(toJSON()).toMatchSnapshot();
  });

  it("should navigate to 'PreviewPostScreen' when click on CTA", () => {
    const {getByText} = render(<HomeEmptyPostList />);

    const button = getByText('Criar post');

    fireEvent.press(button);

    expect(mockedNavigate).toHaveBeenCalledWith('PreviewPostScreen');
    expect(mockedNavigate).toHaveBeenCalledTimes(1);
  });
});
