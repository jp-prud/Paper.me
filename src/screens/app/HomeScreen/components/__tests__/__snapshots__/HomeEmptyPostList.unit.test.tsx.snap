// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeEmptyPostList - unit should render the component 1`] = `
[
  <View
    style={
      [
        {},
      ]
    }
    testID="home-empty-post-list"
  >
    <Text
      selectionColor="#c7e7ea"
      style={
        [
          {
            "color": "#262626",
            "textAlign": "center",
          },
          [
            {
              "fontSize": 16,
              "lineHeight": 22.4,
            },
            {
              "fontFamily": "PlusJakartaSans-Regular",
            },
            undefined,
          ],
        ]
      }
    >
      Nenhum post encontrado.
    </Text>
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        selectionColor="#c7e7ea"
        style={
          [
            {
              "color": "#262626",
              "textAlign": "center",
            },
            [
              {
                "fontSize": 16,
                "lineHeight": 22.4,
              },
              {
                "fontFamily": "PlusJakartaSans-Regular",
              },
              undefined,
            ],
          ]
        }
      >
        Seja o primeiro a criar um post.
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#017582",
            "borderRadius": 16,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          selectionColor="#c7e7ea"
          style={
            [
              {
                "color": "#FFFFFF",
              },
              [
                {
                  "fontSize": 16,
                  "lineHeight": 22.4,
                },
                {
                  "fontFamily": "PlusJakartaSans-Bold",
                },
                undefined,
              ],
            ]
          }
        >
          Criar post
        </Text>
      </View>
    </View>
  </View>,
  " ",
]
`;
